### PR TITLE
JM-6746: Generate attendance audit numbers when confirming pool attendance (jurors in waiting)

### PR DESCRIPTION
**Pull request by CCHAPMAN01**
### JIRA link (if applicable) ###
https://centralgovernmentcgi.atlassian.net/browse/JM-6746

### Change description ###
- update query to fetch attendance records
  - only fetch records with an active juror_pool record
- refactor getAttendanceAuditNumber method to be reusable for jury attendance and pool attendance
- add new helper method to create pool attendance history
- update logic to generate attendance audit number when confirming pool attendance
- update unit and integration tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by Ben-Edwards-cgi — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/JurorHistoryServiceImpl.java [hunk 1]
```diff
@@ -241,6 +241,10 @@ public void createJuryAttendanceHistory(JurorPool jurorPool, String otherInfo) {
         registerHistorySystem(jurorPool, HistoryCodeMod.JURY_ATTENDANCE, otherInfo);
     }
 
+    public void createPoolAttendanceHistory(JurorPool jurorPool, String otherInfo) {
+        registerHistorySystem(jurorPool, HistoryCodeMod.POOL_ATTENDANCE, otherInfo);
```
[the remark this answers is not in the document]
Comment (CCHAPMAN01) [reply src/main/java/uk/gov/hmcts/juror/api/moj/service/JurorHistoryServiceImpl.java:245]: Good spot should be user generated ... as should the one above I created yesterday 🙈 will update both + tests 👍

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java [hunk 2]
```diff
@@ -936,6 +935,44 @@ void updateAttendanceNoShow() {
                 .containsExactlyInAnyOrder(Boolean.TRUE, Boolean.TRUE);
         }
 
+        @Test
+        @DisplayName("PATCH Update attendance - confirm attendance")
+        @Sql({"/db/mod/truncate.sql", "/db/jurormanagement/UpdateAttendanceDetails.sql",
+            "/db/JurorExpenseControllerITest_expenseRates.sql"})
```
[the remark this answers is not in the document]
Comment (CCHAPMAN01) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java:941]: Great will remove

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/jurormanagement/JurorAppearanceServiceImpl.java [hunk 3]
```diff
@@ -252,7 +252,7 @@ public AttendanceDetailsResponse updateAttendance(BureauJwtPayload payload, Upda
 
     @Override
     @Transactional
-    @SuppressWarnings("PMD.LawOfDemeter")
+    @SuppressWarnings({"PMD.LawOfDemeter", "PMD.ConfusingTernary"})
```
[the remark this answers is not in the document]
Comment (CCHAPMAN01) [reply src/main/java/uk/gov/hmcts/juror/api/moj/service/jurormanagement/JurorAppearanceServiceImpl.java:256]: Yeah the actual code wasn't confusing ... but you're right it sets a bad precedent so will invert the logic and remove the suppression

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/jurormanagement/JurorAppearanceServiceImpl.java [hunk 4]
```diff
@@ -776,6 +778,15 @@ AttendanceDetailsResponse updateConfirmAttendance(UpdateAttendanceDto.CommonData
         return response;
     }
 
+    private String getAttendanceAuditNumber(String prefix) {
+
+        // get the next available attendance number from the database sequence
+        final long attendanceAuditNumber = appearanceRepository.getNextAttendanceAuditNumber();
+
+        // left pad with 0's up to 8 digits and add 'J' prefix for Jury Attendance
+        return prefix + String.format("%08d", attendanceAuditNumber);
```
[the remark this answers is not in the document]
Comment (CCHAPMAN01) [reply src/main/java/uk/gov/hmcts/juror/api/moj/service/jurormanagement/JurorAppearanceServiceImpl.java]: added new flyway script to alter the sequence and manage the format/length in the DB as suggested 👍

File: src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (CCHAPMAN01) [reply src/integration-test/java/uk/gov/hmcts/juror/api/moj/controller/JurorManagementControllerITest.java:939]: test failed when removing the set up script 🙈  re-instating it fixed it, so will leave it in for now

File: src/main/java/uk/gov/hmcts/juror/api/moj/service/JurorHistoryServiceImpl.java [hunk 5]
```diff
@@ -237,8 +237,12 @@ public void createSummonsLetterHistory(JurorPool jurorPool, String otherInfo) {
         registerHistorySystem(jurorPool, HistoryCodeMod.PRINT_SUMMONS, otherInfo);
     }
 
-    public void createJuryAttendanceHistory(JurorPool jurorPool, String otherInfo) {
-        registerHistorySystem(jurorPool, HistoryCodeMod.JURY_ATTENDANCE, otherInfo);
+    public void createJuryAttendanceHistory(JurorPool jurorPool, String otherInfo, String username) {
+        registerHistory(jurorPool, HistoryCodeMod.JURY_ATTENDANCE, otherInfo, username);
```
Comment (Ben-Edwards-cgi) [inline src/main/java/uk/gov/hmcts/juror/api/moj/service/JurorHistoryServiceImpl.java:241]: We should have a method to pull the username from SecurityUtil so we don't need to pass it in here

**Pull request merged**